pkg/proxy/iptables/proxier.go

```diff
@@ -823,7 +870,12 @@ func (proxier *Proxier) syncProxyRules() {
 		isIPv6 := utilnet.IsIPv6(svcInfo.ClusterIP())
 		protocol := strings.ToLower(string(svcInfo.Protocol()))
 		svcNameString := svcInfo.serviceNameString
-		hasEndpoints := len(proxier.endpointsMap[svcName]) > 0
+
+		svcEndpoints := proxier.endpointsMap[svcName]
+		if !svcInfo.OnlyNodeLocalEndpoints() && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyAwareServiceRouting) {
```

Comment:
Because `externalTrafficPolicy:Local` should have a higher priority, if it is Local, it only cares about the endpoint on current node, and should not care the topology.

Reply:
Nice catch!